base/applications/taskmgr/graphctl.c

```diff
@@ -12,666 +12,363 @@
-            {
-                RECT rc;
-                rc.bottom = this->m_rectClient.bottom + 1;
-                rc.left = prevX;
-                rc.right = currX + 1;
-                rc.top = this->m_rectPlot.bottom + 1;
-                /* RECT rc(prevX, m_rectPlot.bottom+1, currX+1, m_rectClient.bottom+1); */
-                FillRect(this->m_dcPlot, &rc, this->m_brushBack);
-            }
-
-            /*  store the current point for connection to the next point */
-            this->m_dPreviousPosition[i] = this->m_dCurrentPosition[i];
+            if(!j)  j = inst->NumberOfPoints - 1;
+            else    j--;
+            y = h - *(t + j) * coef;
+            x -= PLOT_SHIFT;
+            LineTo(hdcg, x, y);
         }
     }
-}
+    SelectObject(hdcg, inst->hPen0);
+    t = inst->PointBuffer;
+    x = inst->BitmapWidth - 1;
+    j = inst->CurrIndex;
+    y = h - *(t + j) * coef;
+    MoveToEx(hdcg, x, y, NULL);
+    for (i = 0; i < inst->NumberOfPoints; i++)
+    {
+        j = (j ? j : inst->NumberOfPoints) - 1;
+        y = h - *(t + j) * coef;
+        x -= PLOT_SHIFT;
+        LineTo(hdcg, x, y);
+    }
 
-void GraphCtrl_Resize(TGraphCtrl* this)
-{
-    /*  NOTE: Resize automatically gets called during the setup of the control */
-    GetClientRect(this->m_hWnd, &this->m_rectClient);
-
-    /*  set some member variables to avoid multiple function calls */
-    this->m_nClientHeight = this->m_rectClient.bottom - this->m_rectClient.top;/* m_rectClient.Height(); */
-    this->m_nClientWidth  = this->m_rectClient.right - this->m_rectClient.left;/* m_rectClient.Width(); */
-
-    /*  the "left" coordinate and "width" will be modified in  */
-    /*  InvalidateCtrl to be based on the width of the y axis scaling */
-#if 0
-    this->m_rectPlot.left   = 20;
-    this->m_rectPlot.top    = 10;
-    this->m_rectPlot.right  = this->m_rectClient.right-10;
-    this->m_rectPlot.bottom = this->m_rectClient.bottom-25;
-#else
-    this->m_rectPlot.left   = 0;
-    this->m_rectPlot.top    = -1;
-    this->m_rectPlot.right  = this->m_rectClient.right;
-    this->m_rectPlot.bottom = this->m_rectClient.bottom;
-#endif
-
-    /*  set some member variables to avoid multiple function calls */
-    this->m_nPlotHeight = this->m_rectPlot.bottom - this->m_rectPlot.top;/* m_rectPlot.Height(); */
-    this->m_nPlotWidth  = this->m_rectPlot.right - this->m_rectPlot.left;/* m_rectPlot.Width(); */
-
-    /*  set the scaling factor for now, this can be adjusted  */
-    /*  in the SetRange functions */
-    this->m_dVerticalFactor = (double)this->m_nPlotHeight / this->m_dRange;
+    return;
 }
 
-#if 0
-void TGraphCtrl::Reset()
+inline void
+TmgcRedrawOnHeightChange(PTM_GRAPH_CONTROL inst, INT nh)
 {
-    /*  to clear the existing data (in the form of a bitmap) */
-    /*  simply invalidate the entire control */
-    InvalidateCtrl();
+    HDC     hdc;
+    HBITMAP hbmOld;
+
+    inst->BitmapHeight = nh;
+    inst->ftPixelsPerPercent = (float)nh / 100.00f;
```

Comment:
```suggestion
    inst->ftPixelsPerPercent = (FLOAT)nh / 100.00f;
```

Reply:
why?

Reply:
It followed another identical change elsewhere in the PR.